app/models/custom_hint.rb

```diff
@@ -64,7 +65,13 @@ def validate_csv
   def process_records
```

Comment:
rubocop does flag this method as a bit long and a bit complex. I won't block merge on that as it's not egregious, but consider whether any of this logic could break out to another method to make it easier to read in the future.